polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -5152,15 +5153,16 @@ void MLIRASTConsumer::run() {
     std::string name;
     MLIRScanner::getMangledFuncName(name, FD, CGM);
 
-    if (done.count(name))
-      continue;
-
     LLVM_DEBUG(llvm::dbgs() << "\n-- FUNCTION BEING EMITTED: "
                             << FD->getNameAsString() << " --\n\n";);
 
-    done.insert(name);
     MLIRScanner ms(*this, module, deviceModule, LTInfo);
     auto function = GetOrCreateMLIRFunction(F, true);
+
+    // Skip function if it has already been initialized.
+    if (!done.emplace(F.context, name).second)
+      continue;
+
```

Comment:
This can be move back to line 5155 after PR #6916.

Reply:
Indeed, that's why I was waiting for that change.